bgpd/bgp_routemap.c

```diff
@@ -5110,6 +5161,28 @@ DEFUN (no_match_ipv6_next_hop,
 				      RMAP_EVENT_MATCH_DELETED);
 }
 
+DEFPY (match_ipv4_next_hop,
+       match_ipv4_next_hop_cmd,
+       "[no$no] match ip next-hop address [A.B.C.D]",
+       NO_STR
+       MATCH_STR
+       IP_STR
+       "Match IP next-hop address of route\n"
+       "IP address\n"
+       "IP address of next-hop\n")
```

Comment:
Not sure about the history of this command. Thinking IPv6 was incomplete by providing this option? IPv4 already has option to match nexthp based on prefix-list and access-list. This option can be configured with prefix-list containing one nexthop. Where as IPv6 does not have those options which makes me think that IPv6 was incomplete and not IPv4.

Reply:
I have already got feedbacks from more than one client complaining about the fact that the 'match ip nexthop IP' command was not available.
About your remark then, I would ask for someone else about that. 

here was the comparison:
```
ubuntu1604es(config-route-map)# match ipv6 address 
  WORD         IPv6 access-list name
  prefix-len   Match prefix length of ipv6 address
  prefix-list  Match entries of prefix-lists

ubuntu1604es(config-route-map)# match ipv6 next-hop 
  X:X::X:X  IPv6 address of next hop
  type      Match entries by type


ubuntu1604es(config-route-map)# match ip address 
  (1-199)      IP access-list number
  (1300-2699)  IP access-list number (expanded range)
  WORD         IP Access-list name
  prefix-len   Match prefix length of ip address
  prefix-list  Match entries of prefix-lists

ubuntu1604es(config-route-map)# match ip next-hop 
  (1-199)      IP access-list number
  (1300-2699)  IP access-list number (expanded range)
  WORD         IP Access-list name
  prefix-len   Match prefixlen of given nexthop
  prefix-list  Match entries of prefix-lists
  type         Match entries by type
```

Reply:
Ok, curious to know if customer is interested only in symmetry or is he actually using "match ipv6 address next-hop <ipv6-address>" and want to use the same for IPv4 as well? As the same can be achieved with prefix-list in case of IPv4. I compared both IPv6 and IPv4 and that is why I asked if IPv6 next-hop was a short cut given.

Reply:
if you use prefix-list, your config will be come huger
one prefix list = 3 additional lines + the reference to this.
with filling in on one line the ip, it is simpler.